archive-cleaner

```diff
@@ -68,32 +67,48 @@ def get_packages_from_buildinfo(archive, filename, archive_entry, arch):
     packages = []
     if buildinfo_format > 0:
         for package in installed:
-            pkgfilename = f'{package}.pkg.tar.xz'
+            pkgfilenames = [f'{package}.pkg{ext}' for ext in PKG_EXTENSIONS]
             pkgname = '-'.join(package.split('-')[:-3])
             if not pkgname:
                 print(f'ERROR: failed to parse {package} with format {buildinfo_format} needed for {filename}', file=stderr)
                 continue
-            directory = join(archive, 'packages', pkgname[0], pkgname)
-            path = join(directory, pkgfilename)
-            if not isfile(path):
-                print(f'ERROR: {package} needed for {filename} not in archive {path}', file=stderr)
+            directory = join(archive, 'packages'. pkgname[0], pkgname)
+            paths = [(pkgfilename, join(directory, pkgfilename)) for pkgfilename in pkgfilenames]
```

Comment:
why do we store a tuple here if we only need the path?

Reply:
You need the `pkgfilename` later to add in the `packages` list. I thought this was the simplest solution to keep most of the original code after. Looping via tuples when the existing path is found I can add `pkgfilename` to `packages` as it is the first element of the tuple and then break ì

Reply:
I mean even in the `map` and `filter` approach having both `pkgfilename` and it's path are required, path for filtering existing files and `pkgfilename` to add to `packages` array later, I think tuples are the way of making it two lines like it is

Reply:
it makes it lot harder to read and in mind keep track of what that tuple potentially contains and what not, its way easier to read and understand if `packages` gets extended after a `map` call to `basename`

Reply:
yeah sure I guess that would increase readability thanks